app/decorators/service_provider_session_decorator.rb

```diff
@@ -118,7 +118,9 @@ def url_options
   end
 
   def irs_attempts_api_session_id
-    @irs_attempts_api_session_id ||= request_url_params['irs_attempts_api_session_id']
+    @irs_attempts_api_session_id ||= begin
+      request_url_params['irs_attempts_api_session_id'] || request_url_params['tid']
+    end
```

Comment:
I don't think we need the `begin` ... `end`, the plain operators should have the right precedence here
```suggestion
    @irs_attempts_api_session_id ||=
      request_url_params['irs_attempts_api_session_id'] || request_url_params['tid']
```